### PR TITLE
[d3d11] Implement HDR swapchain features

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Implements CheckColorSpaceSupport, SetColorSpace and SetHDRMetadata.

### CODE REVIEW

File: src/vulkan/vulkan_presenter.h [hunk 1]
```diff
@@ -183,6 +183,17 @@ namespace dxvk::vk {
...
     }
 
+    /**
+     * \brief Gets list of supported surface formats
+     *
+     * \param [out] formats List of supported surface formats
+     * \param [in] fullScreenExclusive Fullscreen exclusive state
+     * \returns Status of the operation
+     */
+    VkResult getSupportedFormats(
```
Comment (doitsujin) [inline src/vulkan/vulkan_presenter.h:194]: Can we expose a `supportsColorSpace` kind of function instead? Having to pass somewhat unrelated data around and also receiving info that we don't need for that purpose (the exact VkFormat) is kind of bad design.

**Review by doitsujin — CHANGES_REQUESTED** [review]
One thing I'd like to see addressed, looks good otherwise.

File: src/vulkan/vulkan_presenter.h (hunk 1 above)
Comment (Joshua-Ashton) [reply src/vulkan/vulkan_presenter.h:194]: Sure

Comment (Joshua-Ashton) [reply src/vulkan/vulkan_presenter.h]: Done.

**Pull request merged**